@pmorie Tested it with DOCKERIZE_KUBELT and e2e for secrets passed. However for emptydir, out of 18, 14 passed and 4 failed.

### Secrets e2e test:

`e2e.test --host="127.0.0.1:8080" --provider="local" --ginkgo.v=true -ginkgo.focus="Secrets" --kubeconfig="/root/.kube/config"`

```
• [SLOW TEST:27.088 seconds]
Secrets
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/secrets.go:100
  should be consumable from pods [Conformance]
  /root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/secrets.go:99
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
Ran 1 of 196 Specs in 27.097 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 195 Skipped PASS
````

### Emptydir  e2e tests (18 tests):
`e2e.test --host="127.0.0.1:8080" --provider="local" --ginkgo.v=true -ginkgo.focus="EmptyDir" --kubeconfig="/root/.kube/config"`

```
Summarizing 4 Failures:

[Fail] EmptyDir volumes [It] volume on tmpfs should have the correct mode using FSGroup [Conformance] [Skipped] 
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/util.go:1118

[Fail] EmptyDir volumes [It] new files should be created with FSGroup ownership when container is non-root [Conformance] [Skipped] 
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/util.go:1118

[Fail] EmptyDir volumes [It] volume on default medium should have the correct mode using FSGroup [Conformance] [Skipped] 
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/util.go:1118

[Fail] EmptyDir volumes [It] new files should be created with FSGroup ownership when container is root [Conformance] [Skipped] 
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/util.go:1118

Ran 18 of 196 Specs in 313.155 seconds
FAIL! -- 14 Passed | 4 Failed | 0 Pending | 178 Skipped --- FAIL: TestE2E (313.16s)
FAIL
```

--- a/hack/local-up-cluster.sh
+++ b/hack/local-up-cluster.sh
@@ -253,7 +253,22 @@ function start_controller_manager {
 
 function start_kubelet {
     KUBELET_LOG=/tmp/kubelet.log
+
+    mkdir -p /var/lib/kubelet
     if [[ -z "${DOCKERIZE_KUBELET}" ]]; then
+      # On selinux enabled systems, it might
+      # require to relabel /var/lib/kubelet
+      if which selinuxenabled &> /dev/null && \
+         selinuxenabled && \
+         which chcon > /dev/null ; then
+         if [[ ! $(ls -Zd /var/lib/kubelet) =~ system_u:object_r:svirt_sandbox_file_t:s0 ]] ; then
+            echo "Applying SELinux label to /var/lib/kubelet directory."
+            if ! chcon -R system_u:object_r:svirt_sandbox_file_t:s0 /var/lib/kubelet; then
+               echo "Failed to apply selinux label to /var/lib/kubelet."
+            fi
+	 fi
+      fi
+
       sudo -E "${GO_OUT}/kubelet" ${priv_arg}\
         --v=${LOG_LEVEL} \
         --chaos-chance="${CHAOS_CHANCE}" \
@@ -277,7 +292,7 @@ function start_kubelet {
         --volume=/var/run:/var/run:rw \
         --volume=/sys:/sys:ro \
         --volume=/var/lib/docker/:/var/lib/docker:ro \
-        --volume=/var/lib/kubelet/:/var/lib/kubelet:rw \
+        --volume=/var/lib/kubelet/:/var/lib/kubelet:rw,z \
         --net=host \
         --privileged=true \
         -i \

